lib/rubocop/cop/performance/count.rb

```diff
@@ -24,13 +24,30 @@ module Performance
       #   [1, 2, 3].count { |e| e < 2 && e.even? }
       #   Model.select('field AS field_one').count
       #   Model.select(:value).count
+      #
+      # Rails compatibility:
+      # Rails will ignore the block that is passed to `count`.
+      # Other methods, such as `select`, will convert the association to an
+      # array and then run the block on the array. A simple work around to
+      # make `count` work with a block is to call `to_a.count {...}`.
+      #
+      # Example:
+      #   Model.where(id: [1, 2, 3].select { |m| m.method == true }.size
+      #
+      #   becomes:
+      #
+      #   Model.where(id: [1, 2, 3]).to_a.count { |m| m.method == true }
       class Count < Cop
         MSG = 'Use `count` instead of `%s...%s`.'.freeze
 
         SELECTORS = [:reject, :select].freeze
         COUNTERS = [:count, :length, :size].freeze
+        RAILS = 'Rails'.freeze
```

Comment:
I'm starting to wonder if we didn't go overboard with this. The resulting code is clearly harder to reader and I don't really care about tiny memory gains.